### PR TITLE
Add geteuid, getgid, getegid tests

### LINKED ISSUE #105: Implement geteuid, getgid, and getegid syscall test
Execution issue for the geteuid, getgid, and getegid portion of #9

### PR DESCRIPTION
**Pull request by ambaxter**
Closes #105

Signed-off-by: Adam Baxter <ambaxter@users.noreply.github.com>

### CODE REVIEW

**Review by connorkuehl — APPROVED** [review]
Thanks, @ambaxter :slightly_smiling_face:

**Review by connorkuehl — CHANGES_REQUESTED** [review]
Actually, sorry, I forgot to check the commit message.

Could you change it to use present-tense imperative and remove the punctuation from the end of the commit subject?

I.e.: "Add geteuid, getgid, getegid tests"

**Comment by ambaxter** [comment]
> Actually, sorry, I forgot to check the commit message.
> 
> Could you change it to use present-tense imperative and remove the punctuation from the end of the commit subject?
> 
> I.e.: "Add geteuid, getgid, getegid tests"

One day I will remember to not use punctuation :D

**Review by connorkuehl — APPROVED** [review]

File: tests/bin/libc.h [hunk 1]
```diff
@@ -145,3 +145,35 @@ uid_t getuid() {
     );
     return rax;
 }
+
+
+uid_t geteuid() {
+    uid_t rax;
+    asm(
```
Comment (haraldh) [inline tests/bin/libc.h:152]: you forgot the clobber registers

**Review by haraldh — CHANGES_REQUESTED** [review]

File: tests/bin/libc.h (hunk 1 above)
Comment (ambaxter) [reply tests/bin/libc.h:152]: Woop! I missed this comment. I'll do this today.

Comment (ambaxter) [reply tests/bin/libc.h:153]: Done

**Review by haraldh — APPROVED** [review]

**Review by connorkuehl — APPROVED** [review]

**Pull request merged**